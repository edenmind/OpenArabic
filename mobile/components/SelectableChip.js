/* eslint-disable import/namespace */
import * as React from 'react'

import { Chip, Text } from 'react-native-paper'

import { COLORS } from '../constants/colors'
import PropTypes from 'prop-types'
import { StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  arabic: {
    fontSize: 17
  },
  chipNotSelected: {
    backgroundColor: COLORS.leaf,
    margin: 5,
    padding: 0
  },
  chipSelected: {
    backgroundColor: COLORS.branch,
    margin: 5,
    padding: 0
  },
  english: {
    fontSize: 23,
    fontWeight: '400'
  }
})

const SelectableChip = (props) => (
  <Chip
    onPress={props.func}
    backgroundColor={COLORS.leaf}
    textStyle={props.language === 'arabic' ? styles.english : styles.arabic}
    style={props.selected ? styles.chipNotSelected : styles.chipSelected}>
    <Text>{props.text ? props.text : 'No text'}</Text>
  </Chip>
)

export default SelectableChip

SelectableChip.propTypes = {
  text: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  func: PropTypes.func.isRequired,
  selected: PropTypes.bool.isRequired
}
