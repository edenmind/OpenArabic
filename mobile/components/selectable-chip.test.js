import { describe, it, jest } from '@jest/globals'
import ShallowRenderer from 'react-test-renderer/shallow' // ES6
import React from 'react'
import renderer from 'react-test-renderer'
import SelectableChip from './selectable-chip.js'

jest.useFakeTimers()
describe('<SnackButton />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SelectableChip text={'abc'} language={'english'} func={() => {}} selected={true} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
