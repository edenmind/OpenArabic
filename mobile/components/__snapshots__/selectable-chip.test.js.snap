// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnackButton /> renders correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "top": undefined,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
  >
    <View
      backgroundColor="rgb(204, 204, 82)"
      collapsable={false}
      style={
        {
          "backgroundColor": "rgb(73, 73, 0)",
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 8,
          "borderStyle": "solid",
          "borderWidth": 1,
          "direction": "ltr",
          "flexDirection": "column",
          "height": 50,
          "margin": 15,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "disabled": false,
            "selected": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            [
              {
                "borderRadius": 8,
              },
              {
                "flexGrow": 1,
              },
            ],
          ]
        }
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "flexGrow": 1,
                "paddingLeft": 4,
                "position": "relative",
              },
              {
                "paddingLeft": 0,
              },
              {
                "paddingRight": 0,
              },
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                [
                  {
                    "marginVertical": 6,
                    "textAlignVertical": "center",
                  },
                  {
                    "color": "rgba(73, 69, 79, 1)",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  {
                    "marginLeft": 16,
                    "marginRight": 16,
                  },
                  {
                    "fontSize": 16,
                  },
                ],
              ]
            }
          >
            abc
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
